### PR TITLE
NAS-132334 / 25.04 / Add environment variables to containers

### PR DESCRIPTION
**Pull request by AlexKarpov98**
**Testing:**

See ticket, result:

https://github.com/user-attachments/assets/34ed18f0-a77c-4ce4-aaf8-d4b5851719ed

### CODE REVIEW

File: src/app/pages/virtualization/components/all-instances/instance-details/instance-general-info/instance-general-info.component.html [hunk 1]
```diff
@@ -21,6 +21,17 @@ <h3 mat-card-title>
       <p>{{ 'Base Image' | translate }}: {{ instance.image.description || '-' }}</p>
       <p>{{ 'CPU' | translate }}: {{ instance.cpu || '-' }}</p>
       <p>{{ 'Memory' | translate }}: {{ formatter.memorySizeFormatting(instance.memory) || '-' }}</p>
+
+      @if (instance.environment && objectKeys(instance.environment).length) {
```
Comment (denysbutenko) [inline src/app/pages/virtualization/components/all-instances/instance-details/instance-general-info/instance-general-info.component.html:25]: hey Alex I just found that `keyvalue` pipe could be used natively

https://github.com/truenas/webui/blob/21722ce6654cb70801715c47f98d67cc81cd9bd9/src/app/pages/apps/components/installed-apps/app-info-card/app-info-card.component.html#L88

**Review by RehanY147 — CHANGES_REQUESTED** [review]
Should be added to the edit form as well

**Comment by AlexKarpov98** [comment]
> Should be added to the edit form as well

In ticket it only says about Creation Form, maybe for edit form there will be separate ticket.

File: src/app/pages/virtualization/components/instance-wizard/instance-wizard.component.html [hunk 2]
```diff
@@ -53,7 +53,35 @@
...
+      <ix-list
+        formArrayName="environmentVariables"
+        [empty]="form.controls.environmentVariables.controls.length === 0"
+        [label]="'Environment Variables' | translate"
+        (add)="addEnvironmentVariable()"
+      >
+        @for (envControl of form.controls.environmentVariables.controls; track envControl; let i = $index) {
+          <ix-list-item
+            [formGroupName]="i"
+            [label]="'Environment Variable ' + (i + 1)"
```
Comment (denysbutenko) [inline src/app/pages/virtualization/components/instance-wizard/instance-wizard.component.html:66]: This label needs `translate` pipe (and one below)

**Review by denysbutenko — APPROVED** [review]
Otherwise looks good for me.

**Review by RehanY147 — APPROVED** [review]

**Pull request merged**